### ISSUE TITLE
On the Spectral Bias of Deep Neural Networks

### ISSUE THREAD

**Issue by Swall0w** [Swall0w/papers#583]
Rahaman, Nasim, Arpit, Devansh, Baratin, Aristide, Draxler, Felix, Lin, Min, Hamprecht, Fred A., Bengio, Yoshua, Courville, Aaron

It is well known that over-parametrized deep neural networks (DNNs) are an
overly expressive class of functions that can memorize even random data with
$100\%$ training accuracy. This raises the question why they do not easily
overfit real data. To answer this question, we study deep networks using
Fourier analysis. We show that deep networks with finite weights (or trained
for finite number of steps) are inherently biased towards representing smooth
functions over the input space. Specifically, the magnitude of a particular
frequency component ($k$) of deep ReLU network function decays at least as fast
as $\mathcal{O}(k^{-2})$, with width and depth helping polynomially and
exponentially (respectively) in modeling higher frequencies. This shows for
instance why DNNs cannot perfectly \textit{memorize} peaky delta-like
functions. We also show that DNNs can exploit the geometry of low dimensional
data manifolds to approximate complex functions that exist along the manifold
with simple functions when seen with respect to the input space. As a
consequence, we find that all samples (including adversarial samples)
classified by a network to belong to a certain class are connected by a path
such that the prediction of the network along that path does not change.
Finally we find that DNN parameters corresponding to functions with higher
frequency components occupy a smaller volume in the parameter.

https://arxiv.org/abs/1806.08734
